eth/precompiles/ecrecover.py

```diff
@@ -25,9 +28,20 @@
 )
 
 
+def validate_data_length(data: bytes, length: int) -> None:
+    if len(data) != length:
+        raise InvalidDataLength(f"Expected data length of {length}. Got {len(data)} instead.")
+
+
 def ecrecover(computation: BaseComputation) -> BaseComputation:
     computation.consume_gas(constants.GAS_ECRECOVER, reason="ECRecover Precompile")
     data = computation.msg.data_as_bytes
+
+    try:
+        validate_data_length(data, 128)
+    except InvalidDataLength:
+        return computation
+
```

Comment:
This is most definitely introducing a consensus issue. See https://github.com/ethereum/go-ethereum/blob/master/core/vm/contracts.go#L167

If the input is smaller than 128, it must be padded with zeroes. If it's larger, just ignore anything that's over the limit.

Reply:
(which the current code already does, afaict)